### ISSUE TITLE
accept tab-complete with enter key?

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-web#15646]
lots of people seem to get tripped up by the fact that if you tab-complete a username, you can't accept the completion by hitting enter, as it sends the message instead

**Comment by t3chguy** [comment]
Related https://github.com/vector-im/element-web/issues/11071

**Comment by mvdan** [comment]
I keep sending messages like `@ma` on public channels for this reason, because the UI showed a popup with the completion to `@martin` and I assumed that pressing enter would complete that. That's how UIs practically everywhere else work - Slack, GitHub, etc.

I don't really mind what the suggested way to use completion should be, but "send a message when the user didn't intend to" seems like a pretty significant bug to me :) I'm going to have to try to remember not to run into the bug for now, but it's going to be hard as the "complete enter key" is in my muscle memory.

**Comment by smithfred** [comment]
The current behaviour is also really annoying for inserting strings of emoji, since the only "select autocompletion" key (that I'm aware of) is <kbd>Space</kbd>. So that means, to insert a string of emoji without spaces via the keyboard, the user has to:

* start typing an autocomplete sequence, e.g. `:smil`;
* select the emoji from the autocomplete list with arrows/tab;
* space to insert (which *also* adds a space);
* start typing another autocomplete sequence *without* deleting the space (since autocomplete doesn't appear without a space after the previous emoji (https://github.com/vector-im/element-web/issues/156350);
* select them from the autocomplete list with arrows/tab;
* space to insert;
* arrow back, delete the first space, arrow forward;
* repeat.

Behaviour should be:

* Enter inserts, not sends.
* The "don't show autocomplete without preceding space" rule should be changed to "don't show unless previous char. is a space or emoji" (or any other character currently allowed).
